tests/_utils/test_stages/gpu.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import annotations
+
+from pathlib import Path
+from subprocess import CompletedProcess
+from typing import Iterator
+
+from .. import DEFAULT_GPU_MEMORY_BUDGET, DEFAULT_GPU_PARALLELISM
+from ..config import Config
+from ..system import ArgList, System
+from .test_stage import TestStage
+
+
+class GPU(TestStage):
+    """A test stage for exercising GPU features.
+
+    Parameters
+    ----------
+    config: Config
+        Test runner configuration
+
+    system: System
+        Process execution wrapper
+
+    """
+
+    kind = "cuda"
+
+    def __init__(self, config: Config, system: System) -> None:
+        self.workers = self.compute_workers(config, system)
+
+    def run(
```

Comment:
the `run` methods from different stages seem to have a lot of overlap, so they could be more aggressively de-duplicated I guess, though they don't seem long enough to really matter. are you expecting a test stage in the future that has a completely different `run` method?

Reply:
Not especially, the reason I kept them distinct was in order to have meaningful names `gpu_args`, `cpu_args` for the generators. We could instead make the generator name generic e.g. `feature_args` and then collapse the `run` methods down. Also some of the stages have custom env and command line args. I would prefer to keep them as-is for now and possibly re-visit in the next PR that will be updating the stages specifically.